crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -389,18 +389,19 @@ def check_repo_tag
     end
 
     def check_key_file
-      expected = @config.fetch("repomd", {})["key_md5"]
+      expected = @config.fetch("repomd", {})["fingerprint"]
       return true if expected.blank?
 
       key_path = repodata_path.join("repomd.xml.key")
       key_path = repodata_media_path.join("repomd.xml.key") unless key_path.file?
 
       if key_path.file?
-        md5 = Digest::MD5.hexdigest(key_path.read)
+        fingerprint = `LC_ALL=C gpg --with-fingerprint #{key_path} | \
+          grep fingerprint | awk -F= '{print $2}'`.split.join(" ")
```

Comment:
I would prefer if we were parsing the output of the gpg command from ruby instead of doing pipes. But not critical.